src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +266,47 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Add any pending ICE candidates from the queue for the client ID
+        Queue<IceCandidate> pendingIceCandidatesQueueByClientId = pendingIceCandidatesMap.get(clientId);
+        while(!pendingIceCandidatesQueueByClientId.isEmpty()) {
+            final IceCandidate iceCandidate = pendingIceCandidatesQueueByClientId.peek();
+            final boolean addIce = localPeer.addIceCandidate(iceCandidate);
```

Comment:
If we failed to add the ice candidate to the `localPeer` essentially we just log that and drop it.  Is there any point in re-trying it?  Just want to make sure it's OK to just drop.  What could cause the `.addIceCandidate` to fail, could it be not ready yet to accept ice candidates or something and we could just "skip" it for now and try again later instead of removing from the queue?  In which case the `while` condition would need to change to simply iterate over the queue once and not check for empty.

Reply:
`.addIceCandidate` would fail if any of the `candidate` members (`sdpMid`, `sdpMLineIndex` or `sdp`) is NULL, in which case there is no point retrying because the candidate is already set. Nevertheless, Google's implementation of the `nativeAddIceCandidate` call is not clear. It is okay to just drop the candidate if the call fails. I would refrain from adding a retry logic at this point because it would make more sense to retry if we can get a clear reason for failure of the API. Since the return type is boolean, it does not give us a lot of information.